src/app/games/siam/tests/siam.component.spec.ts

```diff
@@ -151,15 +142,122 @@ describe('SiamComponent', () => {
         // The the orientation of the piece does not have to be chosen
-        await testUtils.expectClickSuccess('#clickPiece_4_4');
-        const move: SiamMove = new SiamMove(4, 4, MGPOptional.of(Orthogonal.DOWN), Orthogonal.DOWN);
-        await testUtils.expectMoveSuccess('#chooseDirection_DOWN', move);
+        await testUtils.expectClickSuccess('#square_4_4');
+        const move: SiamMove = SiamMove.of(4, 4, MGPOptional.of(Orthogonal.DOWN), Orthogonal.DOWN).get();
+        await testUtils.expectMoveSuccess('#square_4_5', move);
     }));
     it('should toast when clicking as first click on an empty square', fakeAsync(async() => {
         // Given the initial board
         // When clicking on an empty piece
         // Then a toast should say it's forbidden
-        const reason: string = SiamFailure.MUST_INSERT_OR_CHOOSE_YOUR_PIECE();
-        await testUtils.expectClickFailure('#insertAt_2_1', reason);
+        const reason: string = RulesFailure.MUST_CHOOSE_PLAYER_PIECE();
+        await testUtils.expectClickFailure('#square_2_1', reason);
+    }));
+    it('should cancel move when player clicks on the board instead of an orientation arrow', fakeAsync(async() => {
+        // Given a component on which the player has selected a piece for insertion, and a target,
+        // and must select an orientation arrow
+        await testUtils.expectClickSuccess('#remainingPieces_0');
+        await testUtils.expectClickSuccess('#square_0_0');
+        spyOn(testUtils.getComponent(), 'cancelMoveAttempt').and.callThrough();
+        // When clicking somewhere else on the board
+        await testUtils.expectClickSuccess('#square_2_2');
+        // Then it should cancel the move
+        expect(testUtils.getComponent().cancelMoveAttempt).toHaveBeenCalledOnceWith();
+    }));
+    it('should cancel move when player selects an invalid target for insertion (empty)', fakeAsync(async() => {
+        // Given a component on which the player has selected a piece for insertion,
+        // and must select the target for a move
+        await testUtils.expectClickSuccess('#remainingPieces_0');
+        // When the player clicks on an empty target which would result in an impossible move
+        spyOn(testUtils.getComponent(), 'cancelMoveAttempt').and.callThrough();
+        await testUtils.expectClickSuccess('#square_1_1');
+        // Then the move should be canceled
+        expect(testUtils.getComponent().cancelMoveAttempt).toHaveBeenCalledOnceWith();
+    }));
+    it('should cancel move when player selects an invalid target for insertion (mountain)', fakeAsync(async() => {
+        // Given a component on which the player has selected a piece for insertion,
+        // and must select the target for a move
+        await testUtils.expectClickSuccess('#remainingPieces_0');
+        // When the player clicks on an empty target which would result in an impossible move
+        // Then the move should be canceled
```

Comment:
Tiens, les given finaux de rules sont (tous, j'imagine, j'ai pas check) "// Then the move should be " + quelque chose, parmis eux:

53 "illegal"
33 "legal" (dont moultes custom)
9 "deemed illegal"
8 "valid"
6 "forbidden"
6 "accepted"
4 "a success"
3 "deemed legal"
3 "refused"
2 "rejected"
2 "cancelled"
2 "performed" (dont un custom)
1 "considered legal"
1 "juged illegal"
1 "done"

Je propose d'uniformiser sur illegal et legal, should I ticket this :p ?

Reply:
c'est très mineur mais tu peux ajouter au ticket rename yes !